lib/doublemap_api/client/routes.rb

```diff
@@ -4,7 +4,7 @@ class Client::Routes < API
 
     # Return a list of all routes on the system.
     def list
-      get_request('/map/v2/routes').map{ |route| Route.new(route) }
+      get_request('/map/v2/routes').map{ |route| Route.new(route) }.each(&on_route)
```

Comment:
I think I'd prefer `.map{ |route| on_route(Route.new(route)) }` instead of `.map.each` to avoid double iteration. Not like it'll actually make a performance impact, but I think the aesthetic makes more sense.

That does re-introduce the "enforce that it returns model objects" problem, though...